hdate/tekufot.py

```diff
@@ -0,0 +1,265 @@
+"""
+The class attempts to compute:
+    - The tekufot (seasons) in the Hebrew calendar: Nissan, Tammuz, Tishrei, and Tevet.
+    - Calculaltion are based on Shmuel calendar (Talmud Bavli, Eruvin 56a)
+    - Rav Ada’s Tekufah: Closer to the astronomical solar year,
+      shorter by 5min than Schmuel (365 days, 5 hours, 997 parts/chalakim) is not used
+    - Cheilat Geshamim start date, which differs between the diaspora and Israel.
+    - Halachic prayer periods based on key Jewish holidays and seasonal changes.
+    - Appropriate prayer phrases depending on the current date, tradition, and language.
+"""
+
+import datetime
+from enum import IntEnum
+from typing import Union
+
+from hdate import converters as conv
+from hdate.hebrew_date import HebrewDate
+from hdate.location import Location
+from hdate.translator import TranslatorMixin
+from hdate.zmanim import Zmanim
+
+
+class Gevurot(TranslatorMixin, IntEnum):
+    """Enum class for the gevurot."""
+
+    MORID = 0
+    MASHIV = 1
+    NEITHER = 2
+
+
+class Geshamim(TranslatorMixin, IntEnum):
+    """Enum class for the geshamim."""
+
+    BARKHEINU = 0
+    BARECH_ALEINU = 1
+
+
+class Tekufot(TranslatorMixin):  # pylint: disable=too-many-instance-attributes
+    """
+    A class that calculates and manages Jewish seasonal times (Tekufot),
+    periods for prayer insertions, and associated halachic dates such as
+    the start of Cheilat Geshamim (requesting rain)."""
+
+    # pylint: disable=R0913, R0917
+    def __init__(
+        self,
+        date: Union[datetime.date, str, datetime.datetime] = datetime.datetime.now(),
+        diaspora: bool = False,
+        location: Location = Location(),
+        tradition: str = "israel",
+        language: str = "english",
+    ):
+        """Initialize the Tekufot object."""
+        super().__init__()
+        if isinstance(date, str):
+            # If a string is given, parse it into a date/datetime if needed.
+            # For now, assume YYYY-MM-DD format.
+            date = datetime.datetime.strptime(date, "%Y-%m-%d").date()
+        elif isinstance(date, datetime.datetime):
+            date = date.date()
+
+        self.date = date
+        self.gregorian_year = date.year
+        self.hebrew_year = self.gregorian_year + 3760
+        self.diaspora = diaspora
+        self.location = location
+        self.tradition = tradition
+        self.language = language
+
+        # Convert current date to JDN and Hebrew Date
+        self.jdn = conv.gdate_to_jdn(date)
```

Comment:
Do you need all the variables as class variables? jdn/hebrew_date/gregorian_date/hebrew_year/...
I think some of them are unnecessary.

Reply:
ok

Reply:
ok